### PR TITLE
chore: privatize public functions in consul and botocore patch.py

### PR DESCRIPTION
**Pull request by rachelyangdog**
The API should just be what is necessary. A lot of integrations expose implementation details through the patch.py We can’t just remove these functions right away as it will break our API so for each we will need to deprecate first and then remove in 2.x.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by emmettbutler — CHANGES_REQUESTED** [review]
Looks great, thanks for the contribution! Since there are deprecations, this change needs to include a release note.

File: tests/contrib/consul/test_consul_patch.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # removed the ``_generated`` suffix from the file name, to prevent the content
 # from being overwritten by future re-generations.
 
-from ddtrace.contrib.consul import get_version
+from ddtrace.contrib.consul import _get_version
```
Comment (ZStriker19) [inline tests/contrib/consul/test_consul_patch.py:6]:
```suggestion
from ddtrace.contrib.consul.patch import _get_version
```

File: releasenotes/notes/deprecate-patch-py-functions-consul-botocore-da36c9501c7fc279.yaml
Comment (rachelyangdog) [inline releasenotes/notes/deprecate-patch-py-functions-consul-botocore-da36c9501c7fc279.yaml:1]: @emmettbutler release notes!

**Review by emmettbutler — APPROVED** [review]

File: releasenotes/notes/deprecate-patch-py-methods-consul-botocore-da36c9501c7fc279.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+deprecations:
+  - |
+    botocore: All methods in botocore/patch.py except patch() and unpatch() are deprecated and will be removed in version 3.0.0.
```
Comment (ZStriker19) [inline releasenotes/notes/deprecate-patch-py-methods-consul-botocore-da36c9501c7fc279.yaml:4]:
```suggestion
    botocore: All methods in botocore/patch.py except ``patch()`` and ``unpatch()`` are deprecated and will be removed in version 3.0.0.
```

File: releasenotes/notes/deprecate-patch-py-methods-consul-botocore-da36c9501c7fc279.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  - |
+    consul: All methods in consul/patch.py except patch() and unpatch() are deprecated and will be removed in version 3.0.0.
```
Comment (ZStriker19) [inline releasenotes/notes/deprecate-patch-py-methods-consul-botocore-da36c9501c7fc279.yaml:6]:
```suggestion
    consul: All methods in consul/patch.py except ``patch()`` and ``unpatch()`` are deprecated and will be removed in version 3.0.0.
```

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**